Biasing/src/Biasing/TaggerHitFilter.cxx

```diff
@@ -0,0 +1,137 @@
+
+#include "Biasing/TaggerHitFilter.h"
+
+//~~ Geant4 ~~//
+#include "G4RunManager.hh"
+#include "G4Step.hh"
+
+namespace biasing {
+
+TaggerHitFilter::TaggerHitFilter(const std::string& name,
+                                 framework::config::Parameters& parameters)
+    : simcore::UserAction(name, parameters) {
+  layers_hit_ = parameters.getParameter<int>("layers_hit", 8);
+  std::cout << "TaggerHitFilter:: layers_hit_ = " << layers_hit_ << std::endl;
+}
+
+void TaggerHitFilter::stepping(const G4Step* step) {
+  // The track associated with this step will allow for the extraction of info
+  // needed to determine if this is the incident electron.
+  auto track{step->GetTrack()};
+  //  std::cout<<"TaggeHitFilter::stepping"<<std::endl;
+  // Only process the primary electron i.e. a particle without parents.
+  //  June 21, 2024 MG:  Not sure this is a good cut to make here..
+  //                     What about secondaries?
+  //  if (track->GetParentID() != 0) return;
+
+  // Get the PDG ID of the track and make sure it's an electron. If
+  // another particle type is found, thrown an exception because the generator
+  // being used is likely wrong.
+  //  std::cout<<"Track pdgID =
+  //  "<<track->GetParticleDefinition()->GetPDGEncoding()<<std::endl; June 21,
+  //  2024 MG:  similar here, though it's unlikely to be aything else if (auto
+  //  pdgID{track->GetParticleDefinition()->GetPDGEncoding()}; pdgID != 11){
+  //  return;
+  //}
```

Comment:
```suggestion
```
For clarity if you think these comments are not helpful, it's better to remove them